packages/x-date-pickers-pro/src/DateRangePicker/shared.tsx

```diff
@@ -89,6 +89,7 @@ export function useDateRangePickerDefaultizedProps<Props extends BaseDateRangePi
     ...themeProps,
     ...validationProps,
     localeText,
+    enableEnhancedDaySlot: themeProps?.enableEnhancedDaySlot ?? false,
```

Comment:
This default should not be useful right?
Since it's also applied to the Date Range Calendar component.
We should be able to have only the default in `DayCalendar` and maybe `DateRangeCalendar` (since those are the two places where we apply the slot)